### ISSUE TITLE
Better i18n support for cartographic maps

### ISSUE THREAD

**Issue by stevecotton** [wesnoth/wesnoth#5011]
To handle localisation of map images such as the background of the title screen, Wesnoth currently supports overlaying the image with another image with localised names. This has resulted in a rarely-used mechanism for marking the l10n images as fuzzy, without tracking which strings on the map are fuzzy or not (see #4717).

Quoting @GregoryLundberg from https://github.com/wesnoth/wesnoth/issues/4717#issuecomment-578146356
    It requires each translator to be aware a change is required, and have the tools and skills to make the change. It would be far better to eliminate all localization in images and overlay a localized text. This not only places the localization into the normal stream, but also reduces requirements and allows for consistent presentation.

**Comment by stevecotton** [comment]
The map on the title screen is complex, it's shown as a creased sheet of paper which isn't flat - the map bends into the folds in the paper. I think we should accept the current solution is going to be used unchanged in 1.16.

However, it's a much simpler case for maps that aren't creased, for example data/core/images/maps/wesnoth.png.

When map are shown in IntroWML (the `[story]` tag), there's already support for drawing journey-tracking images on top of the map. For the non-creased maps, IntroWML could add support for text boxes.

**Comment by CelticMinstrel** [comment]
A possible compromise might be to have one overlay image per label instead of a single overlay image? And maybe the image could be associated with an actual translated string somehow?

Still, rendering the labels as text would seem to be the ideal option. It might require some advanced logic or features in the text renderer, but I'd say it's definitely the best way to go.

I don't have thetitle  map in front of me, but it might be possible to ignore the creases as long as no label actually crosses a crease.

**Comment by stevecotton** [comment]
> I don't have thetitle map in front of me, but it might be possible to ignore the creases as long as no label actually crosses a crease.

I wondered about that, until I looked more closely at the title map. Like a real sheet of paper, the creases affect the entire sheet, with the full area between the creases being slightly curved. Try zooming in on the map, and then move around so that a horizontal label is near the edge of your viewer's window - the text is on a slight curve.

**Comment by CelticMinstrel** [comment]
Even then I think it could be "close enough" to just draw the text uncurved… or there's also the option of encoding the curve as a text style somehow, though I'm not sure how that would work.

**Comment by stevecotton** [comment]
> When maps are shown in IntroWML (the `[story]` tag), there's already support for drawing journey-tracking images on top of the map. For the non-creased maps, IntroWML could add support for text boxes.

I plan to implement that, and have added it to the roadmap for 1.15.9. I feel it would be useful for UMC, but wouldn't expect it to meet the standards for use in mainline's campaigns.

**Comment by CelticMinstrel** [comment]
Yeah, if you can apply a shear or perspective shift or whatever to get the desired effect while drawing the text as text, that would be an ideal outcome.
